ovh/resource_ip_move.go

```diff
@@ -0,0 +1,309 @@
+		if d.Get("task_status") != nil {
+			ipTask.Status = IpTaskStatusEnum(d.Get("task_status").(string))
+			ipTask.TaskId = taskId
+			taskStartDate, err := time.Parse("2006-01-02 15:04:05 -0700 MST", d.Get("task_start_date").(string))
+			if err != nil {
+				return err
+			}
+			ipTask.StartDate = taskStartDate
+			_, err = waitForTaskFinished(d, meta, ipTask, ip, opts)
+			if err != nil {
+				log.Printf("[WARNING] - waitForTaskFinished on previously registered task return error %s. Will continue nevertheless", err)
+			}
+		}
+	} else {
+		log.Printf("[WARNING] - resource ID %s is not an int64/not a task ID. Cannot get last task state", d.Id())
+	}
+	err = resourceIpServiceReadByServiceName(d, *serviceName, config)
+	if err != nil {
+		return err
+	}
+
+	currentlyRoutedService := GetRoutedToServiceName(d)
+	// no need to update if ip is already routed to the appropriate service
+	if reflect.DeepEqual(currentlyRoutedService, opts.To) {
+		log.Printf("[DEBUG] Won't do anything as ip %s (service name = %s) is already routed to service %v", ip, *serviceName, currentlyRoutedService)
+		return nil
+	} else {
+		if opts.To == nil {
+			log.Printf("[DEBUG] Will move ip %s (service name = %s) from service %s to IP parking", ip, *serviceName, *currentlyRoutedService)
+			endpoint := fmt.Sprintf("/ip/%s/park",
+				url.PathEscape(ip),
+			)
+			// retrieve the task
+			if err := config.OVHClient.Post(endpoint, nil, ipTask); err != nil {
+				return fmt.Errorf("calling Post %s: %q", endpoint, err)
+			}
+		} else {
+			log.Printf("[DEBUG] Will move ip %s (service name = %s) from service %v to service %s", ip, *serviceName, currentlyRoutedService, *opts.To)
+			endpoint := fmt.Sprintf("/ip/%s/move",
+				url.PathEscape(ip),
+			)
+			if err := config.OVHClient.Post(endpoint, opts, ipTask); err != nil {
+				return fmt.Errorf("calling Post %s: %q", endpoint, err)
+			}
+		}
+		d.SetId(fmt.Sprint(ipTask.TaskId))
+		if err = d.Set("task_start_date", ipTask.StartDate.String()); err != nil {
+			return err
+		}
+
+		_, err = waitForTaskFinished(d, meta, ipTask, ip, opts)
+		if err != nil {
+			return err
+		}
+	}
+	return resourceIpRead(d, meta)
+}
+
+// waitForTaskFinished queries GET /ip/:ip/task/:taskId route until task state is in a terminal success or error state or until waitingTimeInSecondsBeforeRefreshState is reached
+// and returns :
+//   - finishedWithSuccess : true if task ended with success, false if ended with error, nil if not ended at all
+//   - err : nil if no error is encountered in the process, any met error otherwise
+//
+// in any case before returning, "task_status" field of d will be updated with the last known ipTask.Status
+func waitForTaskFinished(d *schema.ResourceData, meta interface{}, ipTask *IpTask, ip string, opts *IpMoveOpts) (finishedWithSuccess *bool, err error) {
+	finishedWithSuccess, err = recursiveWaitTaskFinished(d, meta, ipTask, ip, opts)
+	errSet := d.Set("task_status", ipTask.Status)
```

Comment:
we should have a check `ipTask != nil` before this line since it can be modified by the previous call, just to be sure

Reply:
fixed in https://github.com/ovh/terraform-provider-ovh/pull/510/commits/26baa446e1cf64e9bbbdca1c9c12793ab135b497